toml/src/main/kotlin/org/rust/toml/intentions/ExtractDependencySpecificationIntention.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiParserFacade
+import com.intellij.psi.util.parentOfType
+import com.intellij.psi.util.parentsOfType
+import org.rust.cargo.CargoConstants
+import org.rust.lang.core.psi.ext.endOffset
+import org.rust.toml.isDependencyListHeader
+import org.toml.lang.psi.TomlInlineTable
+import org.toml.lang.psi.TomlKeyValue
+import org.toml.lang.psi.TomlPsiFactory
+import org.toml.lang.psi.TomlTable
+
+class ExtractDependencySpecificationIntention : TomlElementBaseIntentionAction<TomlKeyValue>() {
+    override fun getText() = "Extract dependency specification"
+    override fun getFamilyName(): String = text
+
+    override fun findApplicableContextInternal(project: Project, editor: Editor, element: PsiElement): TomlKeyValue? {
+        if (element.containingFile.name != CargoConstants.MANIFEST_FILE) return null
+
+        val dependencyTable = element.parentOfType<TomlTable>()
+            ?.takeIf { it.header.isDependencyListHeader }
+            ?: return null
+
+        val keyValue = element.parentsOfType<TomlKeyValue>()
+            .firstOrNull { it.parent == dependencyTable }
+            ?.takeIf { it.value is TomlInlineTable }
+
+        return keyValue
```

Comment:
`Variable used only in following return and can be inlined`

Also, the name `keyValue` is not very descriptive. WDYT about rewriting it in this way, to indicate that we are trying to find the corresponding `dependency` and then use it only if it contains a table?
```
val dependency = element.parentsOfType<TomlKeyValue>()
    .firstOrNull { it.parent == dependencyTable }

return dependency?.takeIf { it.value is TomlInlineTable }
```